extmod/modopenamp.c

```diff
@@ -181,13 +192,13 @@ static mp_obj_t endpoint_send(uint n_args, const mp_obj_t *pos_args, mp_map_t *k
     for (mp_uint_t start = mp_hal_ticks_ms(); ;) {
         bytes = rpmsg_send_offchannel_raw(&self->ep, src, dest, rbuf.buf, rbuf.len, false);
         if (bytes > 0 || timeout == 0) {
-            MICROPY_EVENT_POLL_HOOK
+            metal_generic_default_poll();
             break;
         }
         if (timeout > 0 && (mp_hal_ticks_ms() - start > timeout)) {
             mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("timeout waiting for a free buffer"));
         }
-        MICROPY_EVENT_POLL_HOOK
+        metal_generic_default_poll();
```

Comment:
This loop, waiting to send, will now be a very busy loop and use lots of CPU power.  It would be better to use `mp_event_wait_ms(1)` here.

And then use `mp_event_handle_nowait()` above just before the `break`.  Then `metal_generic_default_poll()` could continue to call `mp_event_handle_nowait()` (if that `mp_event_handle_nowait` function is ever even used).

(As a side remark, IMO it's better to use the `mp_event_*` functions here in the extmod code because it's well defined what they do, and it makes reasoning about this code easier.  Whereas if `metal_generic_default_poll` is used the behaviour is dependent on the port.)